### PR TITLE
Refactors possible_inscribed_cost to possible_circumscribed_cost in collision checker for smac_planner and mppi_controller

### PR DESCRIPTION
**Pull request by alanxuefei**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (#4113) |
| Primary OS tested on | (Ubuntu, Windows) |
| Robotic platform tested on | (  gazebo simulation ) |
| Does this PR contain AI generated software? | (No) |

---

## Description of contribution in a few bullet points

*  Refactors possible_inscribed_cost to possible_circumscribed_cost in collision checker for smac_planner and mppi_controller
* Ensures text files use LF line endings for consistency, especially for Windows VSCode users.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# This ensures text files use `LF` line endings for consistency, especially for Windows VSCode users.
+* text=auto eol=lf
```
Comment (SteveMacenski) [inline .gitattributes:2]: Please remove this file

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Otherwise looks good to me!

File: .gitattributes (hunk 1 above)
Comment (alanxuefei) [reply .gitattributes]: Removed.

My VSCode runs on Windows.
The files are converted to CRLF line endings when I clone them on Windows without `.gitattributes`.
The following three files cannot be run in the Linux development container of VSCode:

- .devcontainer/on-create-command.sh
- .devcontainer/post-create-command.sh
- .devcontainer/update-content-command.sh

The file format can be seen in the bottom right of VSCode as shown below:
image


If needed, I can add this setting in the future.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Thanks!

**Pull request merged**